extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.instant.js

```diff
@@ -109,30 +109,40 @@ require([
 		}, onPlayerReady);
 	}
 
-	trackingOptIn.pushToUserConsentQueue(function () {
-		doc.addEventListener('bab.blocking', function () {
-			if (adContext.get('opts.wadHMD')) {
-				hmdRecLoader.setOnReady(function () {
-					setupPlayer();
-				});
-			} else {
+	function prePlayerSetup(blocking) {
+		if (blocking && adContext.get('opts.wadHMD')) {
+			hmdRecLoader.setOnReady(function () {
 				setupPlayer();
-			}
-		});
+			});
 
-		doc.addEventListener('bab.not_blocking', function () {
-			if (bidders && bidders.isEnabled()) {
-				bidders.runOnBiddingReady(function () {
-					bidParams = bidders.updateSlotTargeting(featuredVideoSlotName);
-					setupPlayer();
-				});
-			} else {
+			return;
+		}
+
+		if (!blocking && bidders && bidders.isEnabled()) {
+			bidders.runOnBiddingReady(function () {
+				bidParams = bidders.updateSlotTargeting(featuredVideoSlotName);
 				setupPlayer();
-			}
-		});
+			});
 
-		if (!adContext.get('opts.showAds')) {
+			return;
+		}
+
+		setupPlayer();
```

Comment:
It's getting hard to see what's blocking the player setup and in what order. What do you think about calling it in one place and blocking with all dependencies? Something like:
```suggestion
function waitForBab() {
	if (!adContext.get('opts.babDetectionDesktop')) {
		return Promise.resolve();
	}

	return new Promise(function (resolve, reject) {
		doc.addEventListener('bab.blocking', function () {
			reject();
		});
		doc.addEventListener('bab.not_blocking', function () {
			resolve();
		});
	});
}

function waitForRec() {
	if (!adContext.get('opts.wadHMD')) {
		return Promise.resolve();
	}
	
	return new Promise(function (resolve) {
		hmdRecLoader.setOnReady(function () {
			resolve();
		});
	});
}

function waitForBidders() {
	if (!bidders || !bidders.isEnabled()) {
		return Promise.resolve();
	}
	
	return new Promise(function (resolve) {
		bidders.runOnBiddingReady(function () {
			bidParams = bidders.updateSlotTargeting(featuredVideoSlotName);
			resolve();
		});
	});
}

trackingOptIn.pushToUserConsentQueue(function () {
	if (adContext.get('opts.showAds')) {
		waitForBab()
			.then(waitForBidders, waitForRec)
			.then(setupPlayer);
	} else {
		setupPlayer();
	}
});
```
What do you think?

Reply:
:ie11: